newdle/client/src/components/summary/SummaryPage.js

```diff
@@ -229,88 +212,130 @@ export default function SummaryPage() {
-                    checked={sendDeletionMail}
-                    onChange={() => setSendDeletionMail(!sendDeletionMail)}
-                    label="Notify the participants via e-mail"
-                  />
-                  {sendDeletionMail && (
-                    <Input
-                      type="text"
-                      placeholder={t`Leave a comment (optional)`}
-                      className={styles['deletion-comment']}
-                      value={comment}
-                      onChange={(__, {value}) => setComment(value)}
-                    />
-                  )}
-                  {hasParticipantsWithoutEmail && (
-                    <Message className={styles['email-participant-list']}>
-                      <strong>
-                        <Trans>
-                          Some of your participants do not have e-mail addresses and will not be
-                          contacted:
-                        </Trans>
-                      </strong>
-                      <RecipientList
-                        recipients={participantsWithoutEmail}
-                        color="red"
-                        icon="close"
-                      />
-                    </Message>
-                  )}
-                </div>
-              )}
-            </Modal.Content>
-            <Modal.Actions>
+            <div className={styles['button-row']}>
               <Button
-                negative
-                onClick={() => handleDeletionModalConfirm(true)}
-                loading={deletionMailSending || deleting}
+                type="submit"
+                loading={submitting}
+                disabled={!finalDate}
+                className={styles['finalize-button']}
+                onClick={update}
               >
-                <Trans>Delete newdle</Trans>
+                <Trans>Select final date</Trans>
               </Button>
-            </Modal.Actions>
-          </Modal>
+              <Dropdown
+                text={t`Edit`}
+                floating
+                button
+                direction="right"
+                color="violet"
+                as={Button}
+                className={styles['edit-button']}
+              >
+                <Dropdown.Menu>
+                  <Dropdown.Item as={Link} text={t`Participants`} to={`${editUrl}/participants`} />
+                  <Dropdown.Item as={Link} text={t`Timeslots`} to={editUrl} />
+                  <Dropdown.Item as={Link} text={t`Options`} to={`${editUrl}/options`} />
+                  <Dropdown.Item
+                    text={t`Delete newdle`}
+                    onClick={() => setDeletionModalOpen(true)}
+                  />
+                </Dropdown.Menu>
+              </Dropdown>
+            </div>
+          )}
+          <DeleteModal open={deletionModalOpen} setOpen={setDeletionModalOpen} />
         </>
       )}
     </Container>
   );
 }
+
+export function DeleteModal({open, setOpen}) {
```

Comment:
I think this component is big enough to go in its own .js file

Reply:
And it is worth having it separated when it's only used here?

Reply:
Avoiding huge files with multiple large components...

I mean I like putting small 'local' components in the same file because it's more convenient but when things get large I like e.g. the ability to open component A on the left side of my editor and component B on the right side (and opening the same file twice is awkward ;))